### PR TITLE
Mm 34578 add subscribe now button

### PR DESCRIPTION
**Pull request by pablovelezvidal**
#### Summary
Add the subscribe now button to the main menu

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-34578

#### Screenshots
Captura de pantalla 2021-04-14 a las 20 15 44

### CODE REVIEW

**Comment by marianunez** [comment]
@pablovelezvidal we can change the target branch to `master` given that we will not include it in this first phase release - we are cutting too late to cherry-pick this change to `cloud`

**Comment by annelieseklein** [comment]
@pablovelezvidal looks good - the background should have rounded corners and be 8px away from the top, left and right borders of the dropdown menu. "Subscribe Now" should also have no underline

**Comment by pablovelezvidal** [comment]
Captura de pantalla 2021-04-14 a las 23 34 38


> @pablovelezvidal looks good - the background should have rounded corners and be 8px away from the top, left and right borders of the dropdown menu. "Subscribe Now" should also have no underline

ok, I think are fixed now. Can you please verify?

**Comment by annelieseklein** [comment]
@pablovelezvidal it's not showing up for me on the test server

**Comment by marianunez** [comment]
> @pablovelezvidal it's not showing up for me on the test server

We won't be able to test on the PR until we merge the portal changes as well. This will be available soon. Will ping back to take a look.

File: components/main_menu/main_menu.jsx [hunk 1]
```diff
@@ -182,6 +185,23 @@ class MainMenu extends React.PureComponent {
                 id={this.props.id}
                 ariaLabel={formatMessage({id: 'navbar_dropdown.menuAriaLabel', defaultMessage: 'main menu'})}
             >
+                {isFreeTrial &&
+                    <Menu.Group>
```
Comment (marianunez) [inline components/main_menu/main_menu.jsx:189]: We only want to show this if `isCloud` too

File: components/main_menu/index.jsx [hunk 2]
```diff
@@ -70,6 +71,10 @@ function mapStateToProps(state) {
     const joinableTeams = getJoinableTeamIds(state);
     const moreTeamsToJoin = joinableTeams && joinableTeams.length > 0;
     const rhsState = getRhsState(state);
+    const isCloud = getLicense(state).Cloud === 'true';
+    const subscription = state.entities.cloud.subscription;
```
Comment (marianunez) [inline components/main_menu/index.jsx:75]: Can you remind me if we are loading this at initial load or if we need to wrap this in the `WithSubscription`?

File: components/main_menu/main_menu.jsx [hunk 3]
```diff
@@ -182,6 +185,23 @@ class MainMenu extends React.PureComponent {
...
+                        <SystemPermissionGate permissions={Permissions.SYSCONSOLE_READ_PERMISSIONS}>
```
Comment (marianunez) [inline components/main_menu/main_menu.jsx:190]: We have a specific permission for the billing section in the System console with `SYSCONSOLE_WRITE_BILLING`
```suggestion
                        <SystemPermissionGate permissions={Permissions.SYSCONSOLE_WRITE_BILLING}>
```

**Review by marianunez — CHANGES_REQUESTED** [review]
Looks great! Can you add a test to cover when it should/shouldn't show?

File: components/main_menu/index.jsx (hunk 2 above)
Comment (pablovelezvidal) [reply components/main_menu/index.jsx:75]: btw, it is already wrapped with the withSubscription hoc.

Comment (marianunez) [reply components/main_menu/index.jsx:75]: Oh, didn't notice that it was already there. Thanks for the clarification!

File: components/main_menu/main_menu.test.jsx [hunk 4]
```diff
@@ -163,6 +163,20 @@ describe('components/Menu', () => {
...
     });
 
+    test('should show subscribe now button when in trial period', () => {
+        const props = {...defaultProps, isCloud: true, isFreeTrial: true};
+        const wrapper = getMainMenuWrapper(props);
+
+        expect(wrapper.find('UpgradeLink')).toHaveLength(1);
+    });
+
+    test('should hide the subscribe now button when NOT in trial period', () => {
```
Comment (marianunez) [inline components/main_menu/main_menu.test.jsx:173]: Can we add one more that it should not show the button if cloud is false ?

**Review by marianunez — APPROVED** [review]
Looks great, thanks @pablovelezvidal! Non-blocking comment below

File: components/main_menu/main_menu.test.jsx (hunk 4 above)
Comment (pablovelezvidal) [reply components/main_menu/main_menu.test.jsx:173]: done!

**Review by nickmisasi — APPROVED** [review]
LGTM!

**Review by annelieseklein — APPROVED** [review]
Looks great!

**Pull request merged**